app/helpers/react_on_rails_helper.rb

```diff
@@ -90,10 +90,18 @@ def define_render_if_dom_node_present(react_component, data_variable, dom_id, tr
     <<-JS.strip_heredoc
       var renderIfDomNodePresent = function() {
         var domNode = document.getElementById('#{dom_id}');
+
         if (domNode) {
           #{debug_js(react_component, data_variable, dom_id, trace)}
-          var reactComponent = #{react_component}(#{data_variable});
-          React.render(reactComponent, domNode);
+          var element;
+
+          if (Object.getPrototypeOf(#{react_component}) === React.Component) {
```

Comment:
:+1:  Looks excellent. CC: @alexfedoseev @samnang. This looks like a good way to determine if the component is a React.Component. We need to make sure this works for createClass as well as the ES6 class definition. Another option could be to check if the object has a property called `render` that's a function. We also need to consider if the React API might change in ways that would break this.